coder-chris/recon_tools
# Clean up Rubocop and add rake task for Asciidoc generation

This PR contains two main changes

* updating and configuring rubocop: Instead of fixing all the violations I generated a `.rubocop-todo.yml`, so that we can discuss witch conventions to follow
* adding a task to generate HTML from asciidoc: This is not a full website generation, but merely creates all the individual HTML files under `out/docs`, so that one can get a quick look.

### How to verify

1. Check out this branch to your machine
2. Run `bundle install` to ensure all gems are present in the required version
3. Run `bundle exec rubocop` and observe that no violations are found. This is because all of them are listed in the `.rubocop-todo.yml` and are ignored until we remove them from that file.
4. Run `rake doc` and observe that files are created under `out/docs`.
5. Open `out/docs/01_intro/020_architecture.html` to see that the HTML generation and the diagram generation was successful.
